toolkit/tools/imagegen/configuration/systemconfig_test.go

```diff
@@ -139,6 +141,121 @@ func TestShouldFailParsingBadUserUID_SystemConfig(t *testing.T) {
 	assert.Equal(t, "failed to parse [SystemConfig]: failed to parse [User]: invalid value for UID (-2), not within [0, 60000]", err.Error())
 }
 
+func TestShouldFailToParsingMultipleSameMounts_SystemConfig(t *testing.T) {
+	var checkedSystemConfig SystemConfig
+
+	badPartitionSettingsConfig := validSystemConfig
+	badPartitionSettingsConfig.PartitionSettings = []PartitionSetting{
+		{MountPoint: "/"},
+		{MountPoint: "/"},
+	}
+
+	err := badPartitionSettingsConfig.IsValid()
+	assert.Error(t, err)
+	assert.Equal(t, "invalid [PartitionSettings]: duplicate mount point found at '/'", err.Error())
+
+	err = remarshalJSON(badPartitionSettingsConfig, &checkedSystemConfig)
+	assert.Error(t, err)
+	assert.Equal(t, "failed to parse [SystemConfig]: invalid [PartitionSettings]: duplicate mount point found at '/'", err.Error())
+}
+
+func TestShouldFailParsingBothVerityAndEncryption_SystemConfig(t *testing.T) {
+	var checkedSystemConfig SystemConfig
+
+	badBothEncryptionVerityConfig := validSystemConfig
+	badBothEncryptionVerityConfig.ReadOnlyVerityRoot = validSystemConfig.ReadOnlyVerityRoot
+	badBothEncryptionVerityConfig.ReadOnlyVerityRoot.Enable = true
+
+	err := badBothEncryptionVerityConfig.IsValid()
+	assert.Error(t, err)
+	assert.Equal(t, "invalid [ReadOnlyVerityRoot] & [Encryption]: verity root currently does not support root encryption", err.Error())
+
+	err = remarshalJSON(badBothEncryptionVerityConfig, &checkedSystemConfig)
+	assert.Error(t, err)
+	assert.Equal(t, "failed to parse [SystemConfig]: invalid [ReadOnlyVerityRoot] & [Encryption]: verity root currently does not support root encryption", err.Error())
+}
+
+func TestShouldFailParsingInvalidVerityRoot_SystemConfig(t *testing.T) {
```

Comment:
These tests belong in a future PR, should we move?

Reply:
If they are failing now - please do move them.

---
In reply to: [547510671](https://github.com/microsoft/CBL-Mariner/pull/479#discussion_r547510671) [](ancestors = 547510671)

Reply:
Ok, I'll see if I can find a way to excise them without breaking all my other commits.